### PR TITLE
Difficulty Panel

### PR DESCRIPTION
**Pull request by Gregory711**
## Checklist for completing pull request:
- [x] Test functionality on Mobile, Web, and Desktop
- [x] Verify that any tests for new functionality are created and functional.
- [x] If needed, update the Readme

Replaced existing difficulty selection buttons with new card based ones. Added difficulty descriptions and stars with symbolic colors and number of points respectively. Removed start game button and updated logic so just clicking on difficulty card starts game of that difficulty. Added two new end to end test suites, one to test functionality of starting games of varying difficulties with the buttons and one to test the new page resizing logic.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `DifficultyPanel` component for selecting game difficulty in Sudoku.
	- Enhanced `PlayPage` layout with a `ScrollView` for better usability on smaller screens.
	- Improved navigation and interaction design for difficulty selection.

- **Bug Fixes**
	- Adjusted responsive design elements for better visibility of game components on various screen sizes.

- **Tests**
	- Added new test cases to verify the game functionality for different difficulty levels and responsive behavior across devices.

### CODE REVIEW

**Comment by Gregory711** [comment]
https://github.com/coderabbitai review

**Comment by Gregory711** [comment]
@coderabbitai review

File: app/Components/Home/DifficultyPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  rowCount: number,
+  cardHeight: number,
+  shrinkage: number
+): number {
+  let fromCards: number = rowCount * cardHeight;
+  let fromPadding: number = (rowCount - 1) * CARD_PADDING;
+  return (fromCards + fromPadding) * (1 - shrinkage);
+}
+
+const DifficultyPanel = (props: any) => {
```
Comment (Thomas-Gallant) [inline app/Components/Home/DifficultyPanel.tsx:74]: nit, can we have typed props?

**Review by Thomas-Gallant — CHANGES_REQUESTED** [review]

File: app/Pages/PlayPage.tsx [hunk 2]
```diff
@@ -121,89 +111,15 @@ const PlayPage = () => {
...
-              style={{ margin: newSize / 4, width: 200 }}
-              buttons={[
-                {
-                  value: "grandmaster",
-                  label: "grandmaster",
-                }, // note: could add random difficulty especially if end design requires even number
-              ]}
-            />
+            <View style={{ alignItems: "center", alignSelf: "center" }}>
+              <View style={{ padding: reSize / 4 }}>
```
Comment (Thomas-Gallant) [inline app/Pages/PlayPage.tsx:115]: Can these be condensed into one view.

**Review by Thomas-Gallant — APPROVED** [review]

**Pull request merged**